pkg/sync/provider.go

```diff
@@ -12,6 +13,37 @@ const (
 	NativeStateMode = "secret"
 )
 
+// VerifySignaturesMode represents the strategy to use when choosing which commits to GPG-verify between the flux sync tag and the tip of the flux branch
+type VerifySignaturesMode string
+
+const (
+	// VerifySignaturesModeNone (default) - don't verify any commits
+	VerifySignaturesModeNone = "none"
+
+	// VerifySignaturesModeAll - consider all possible commits
+	VerifySignaturesModeAll = "all"
+
+	// VerifySignaturesModeFirstParent - consider only commits on the chain of
+	// first parents (i.e. don't consider commits merged from another branch)
+	VerifySignaturesModeFirstParent = "first-parent"
+)
+
+// ToVerifySignaturesMode converts a string to a VerifySignaturesMode
+func ToVerifySignaturesMode(s string) (VerifySignaturesMode, error) {
+	switch s {
+	case "":
+		return VerifySignaturesModeNone, nil
+	case VerifySignaturesModeNone:
+		return VerifySignaturesModeNone, nil
+	case VerifySignaturesModeAll:
+		return VerifySignaturesModeAll, nil
+	case VerifySignaturesModeFirstParent:
+		return VerifySignaturesModeFirstParent, nil
+	default:
+		return VerifySignaturesModeNone, fmt.Errorf("%s is not a valid git-verify-signatures-mode", s)
```

Comment:
Nitpick for observability in logs

```suggestion
		return VerifySignaturesModeNone, fmt.Errorf("'%s' is not a valid git-verify-signatures-mode", s)
```